File: GC.Domain/Gardens/Appeals/Appeal.cs
using System;

namespace GC.Domain.Gardens.Appeals
{
    public class Appeal
    {
        public Guid Id { get; }
        public String FirstName { get; }
        public String LastName { get; }
        public String PhoneNumber { get; }
        public String Title { get; }
        public String Message { get; }
        public String Email { get; }
        public DateTime Date { get; }
        public Boolean IsViewed { get; }

        public Appeal(Guid id, String firstName, String lastName, String phoneNumber, String title, String message, String email,
            DateTime date, Boolean isViewed)
        {
            Id = id;
            FirstName = firstName;
            LastName = lastName;
            PhoneNumber = phoneNumber;
            Title = title;
            Message = message;
            Email = email;
            Date = date;
            IsViewed = isViewed;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/gardens/appeals/appeal.ts
export class Appeal {
    constructor(
        public readonly id: string,
        public readonly firstName: string,
        public readonly lastName: string,
        public readonly phoneNumber: string,
        public readonly title: string,
        public readonly message: string,
        public readonly email: string | null,
        public readonly date: Date,
        public readonly isViewed: boolean,
    ) { }

    public get fullName() {
        return `${this.lastName} ${this.firstName}`;
    }
}

export const mapToAppeal = (data: any) => {
    return new Appeal(data.id, data.firstName, data.lastName, data.phoneNumber, data.title, data.message,
        data.email, new Date(data.date), data.isViewed);
}

export const mapToAppeals = (data: any[]) => {
    return data.map(mapToAppeal);
}